src/movies.js

```diff
@@ -1,19 +1,59 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  return moviesArray.map(function(movie){
+    return movie.director 
+  })  
+  //return myDirectors
+} 
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
-
+function howManyMovies(moviesArray) {
+  const result1 = moviesArray.filter(function(movie0){
+    return movie0.director === 'Steven Spielberg' && movie0.genre.includes('Drama')
+  });
+  return result1.length
+}
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  
+  if (moviesArray.length === 0) {
+    return 0;
+  }
+  const totalScores = moviesArray.reduce(function(accu, movie){
+    if (movie.score > 0) {
+      return accu + movie.score;
+    } else  {
+      return accu ;
+    }
+  }, 0);
+  const average = totalScores / moviesArray.length;
+  const decimal = (value) => Math.round(value * 100) / 100;
```

Comment:
`Math.round()` will take a value and round it to the closest integer, that is been used just to transform the string into a number. When you  `*100` and after `/100` you get the decimals value.  So If you get `Math.round( 6.96 * 100)` you get 696 and if we get `696 /100` we would get the 6.96 as a number  

```
let score = "6.96"

console.log(Math.round(score)) // 7
console.log(Math.round(score * 100)) // 696
console.log(Math.round(score * 100) / 100) // 6.96
```